plugins/module_utils/entrust_cagw/api.py

```diff
@@ -0,0 +1,347 @@
+class CAGWSession(object):
+    def __init__(self, name, **kwargs):
+        """
+        Initialize our session
+        """
+
+        self._set_config(name, **kwargs)
+
+    def client(self):
+        resource = Resource(self)
+        return resource
+
+    def _set_config(self, name, **kwargs):
+        headers = {
+            "Content-Type": "application/json",
+            "Connection": "keep-alive",
+        }
+        self.request = Request(headers=headers, timeout=120)
+
+        configurators = [self._read_config_vars]
+        for configurator in configurators:
+            self._config = configurator(name, **kwargs)
+            if self._config:
+                break
+        if self._config is None:
+            raise SessionConfigurationException(to_native("No Configuration Found."))
+
+
+        # set up client certificate if passed (support all-in one or cert + key)
+        cagw_api_cert = self.get_config("cagw_api_cert")
+        cagw_api_cert_key = self.get_config("cagw_api_cert_key")
+        if cagw_api_cert:
+            self.request.client_cert = cagw_api_cert
+            if cagw_api_cert_key:
+                self.request.client_key = cagw_api_cert_key
+        else:
+            raise SessionConfigurationException(to_native("Client certificate for authentication to the API must be provided."))
+
+        # set up the spec
+        cagw_api_specification_path = self.get_config("cagw_api_specification_path")
+
+        if not cagw_api_specification_path.startswith("http") and not os.path.isfile(cagw_api_specification_path):
+            raise SessionConfigurationException(to_native("OpenAPI specification was not found at location {0}.".format(cagw_api_specification_path)))
+        if not valid_file_format.match(cagw_api_specification_path):
+            raise SessionConfigurationException(to_native("OpenAPI specification filename must end in .json, .yml or .yaml"))
+
+        self.verify = True
+
+        if cagw_api_specification_path.startswith("http"):
+            try:
+                http_response = Request().open(method="GET", url=cagw_api_specification_path)
+                http_response_contents = http_response.read()
+                if cagw_api_specification_path.endswith(".json"):
+                    self._spec = json.load(http_response_contents)
+                elif cagw_api_specification_path.endswith(".yml") or cagw_api_specification_path.endswith(".yaml"):
+                    self._spec = yaml.safe_load(http_response_contents)
+            except HTTPError as e:
+                raise SessionConfigurationException(to_native("Error downloading specification from address '{0}', received error code '{1}'".format(
+                    cagw_api_specification_path, e.getcode())))
+        else:
+            with open(cagw_api_specification_path) as f:
+                if ".json" in cagw_api_specification_path:
```

Comment:
Shouldn't you use `endswith()` instead of `in`?
```suggestion
                if cagw_api_specification_path.endswith(".json"):
```
Otherwise this will also match if `cagw_api_specification_path == '/foo.jsonpath/bla.yml'`.